File: example-projects/testng/src/test/java/com/tngtech/jgiven/example/projects/testng/WhenStage.java
package com.tngtech.jgiven.example.projects.testng;

import com.tngtech.jgiven.Stage;
import com.tngtech.jgiven.annotation.ScenarioState;

class WhenStage extends Stage<WhenStage> {
    @ScenarioState(required = true)
    String message;

    @ScenarioState
    String result;

    public void handle_message() {
        result = message + " TestNG!";
    }
}

File: example-projects/kotlin/src/test/kotlin/com/tngtech/jgiven/example/projects/junit5/WhenStage.kt
package com.tngtech.jgiven.example.projects.junit5

import com.tngtech.jgiven.Stage
import com.tngtech.jgiven.annotation.ScenarioState

open class WhenStage : Stage<WhenStage>() {

    @ScenarioState(required = true)
    lateinit var message: String

    @ScenarioState
    lateinit var result: String

    open fun handle_message() {
        result = "$message 5!"
    }
}
